File: responses/4T59SN-Z8Q/find_maximum_product.js
// you can write to stdout for debugging purposes, e.g.
// console.log('this is a debug message');

function solution(X, B) {
    // write your code in JavaScript
    if(B.length==0 || B.length<X) return 0;

    var maxProduct=0;

    for(var i=0; i<B.length-X; i++){
           var product=1;
           for(var j=1; j<=X; j++){
               product*=B[i+j];
           }
           if(product>maxProduct) maxProduct=product;
    }
    return maxProduct;
}

File: responses/PKEQCM-UXF/find_maximum_product.py
# you can write to stdout for debugging purposes, e.g.
# print("this is a debug message")

def solution(X, B):
    # write your code in Python
    if len(B) == 0 or X > len(B):
        return 0
    else:
        product = 0
        for i in range(len(B)-X+1):
            slice_ = B[i:i+X]
            slice_prod = B[i]
            for j in range(i+1, i+X):
                slice_prod = slice_prod * B[j]

            if slice_prod > product:
                product = slice_prod
        return product